### PR TITLE
Fix getIn unexpected value

### PR DESCRIPTION
**Pull request by pieplu**
As described in #3181

It scares me a bit, touching getIn seems a big change.
But this will correct a bug, rare, but a bug in my opinion

### CODE REVIEW

**Comment by johnrom** [comment]
This seems like a good change to me, but we'll definitely want thorough testing when modifying this particular bit of code.

File: packages/formik/src/utils.ts [hunk 1]
```diff
@@ -66,18 +66,24 @@ export function getActiveElement(doc?: Document): Element | null {
...
-export function getIn(
-  obj: any,
-  key: string | string[],
-  def?: any,
-  p: number = 0
-) {
-  const path = toPath(key);
-  while (obj && p < path.length) {
-    obj = obj[path[p++]];
+ export function getIn(
```
Comment (johnrom) [inline packages/formik/src/utils.ts:69]: Looks like the formatting got messed up here. Is this indented correctly?

Comment (pieplu) [reply packages/formik/src/utils.ts]: You're right, prettier doesn't seem to run automatically. I just fixed it, and added more tests.

**Comment by pieplu** [comment]
up

**Comment by pieplu** [comment]
We have been using this patch in our application, on the test environment, for a few weeks, no regression found yet.
We have about 100 different components using Formik.

I totally understand that there is a fear with a change in such an important part of Formik, and that it takes time. I will keep you posted when we have deployed this to our production environment.

**Comment by pieplu** [comment]
We have been using this patch in our production environment for almost a month.
We have at least 40 users, who use our form components to create and update content. All day, every day.
No regression has been seen, and the related bug no longer occurs.

@jaredpalmer

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**